### DISCUSSION TITLE
Graphics mode bug (missing pixel 0)

### DISCUSSION THREAD

**Discussion by ixmeins** [picocomputer/community#14] [Q&A]
I've been writing some graphics code on the 6502 and I believe I have found a possible bug in the graphics video generation.

If I write 0x0f to address 0 of the screen memory I should have a white pixel at 0,0 and I get nothing which I have confirmed on my Tektronix 500MHz scope. If I instead write 0xf0 to address 0 (pixel 1,0) I get a white pixel and I can see it on the CRO.

It looks like the first (left most) pixel is not being rendered.

**Comment by rumbledethumps** [comment]
[Report it here.](https://github.com/picocomputer/rp6502/issues) It's an easy fix, but I can't do it right away.

**Reply by ixmeins** [reply]
I'll have a wade through the render code (probably good for me on a Sunday long weekend!

**Reply by rumbledethumps** [reply]
hint: https://github.com/picocomputer/rp6502/blob/016eff0884aad55e20510aaeb308d7e18f42ea4a/src/vga/vga.c#L390

Note that graphics code is far from final. This was just enough to allocate resources and prove the hardware works.

**Reply by ixmeins** [reply]
Yep understood - just playing :-)
https://github.com/ixmeins/picocomputer-6502-sprite-editor

**Reply by ixmeins** [reply]
When I rebuild the RIA and VGA code I get a working RIA but a non-working (compiles fine) VGA. The resulting uf2 is also about 40kb shorter than any working version so I'll have to figure out my build env before I can try the fixed render code (++ too early).

**Reply by ixmeins** [reply]
Pixel 0 is now working when I build a vga.uf2 and 180 line mode works (renders the 4bpp buffer) but 240 line mode just reverts to colour bars.
